selfdrive/car/gm/carcontroller.py

```diff
@@ -107,13 +107,23 @@ def update(self, CC, CS):
       if self.CP.networkLocation == NetworkLocation.gateway and self.frame % self.params.ADAS_KEEPALIVE_STEP == 0:
         can_sends += gmcan.create_adas_keepalive(CanBus.POWERTRAIN)
 
+    elif self.CP.networkLocation == NetworkLocation.fwdCamera:
+      # Stock longitudinal, integrated at camera
+      if self.frame % 3 == 0:
```

Comment:
we want to send button presses immediately, especially the cancel. also this seems pretty fast - is this rate necessary?

Reply:
This is the stock rate of 33hz and it takes 100-200 ms to cancel, we'll try lower though.

Reply:
Sending immediately at 10Hz, these are the number of frames to cancel cruise (at 100Hz): 23, 21, 11, 23, 13, 11, 16.

Sometimes it only cancels after we send a single button message, so the best we can do is 10 openpilot frames (100 ms).